### PR TITLE
Fix length problem in SSL Cosine LR

### LINKED ISSUE #636: Simclr cosine LR scheduler shows wrong behaviour when linear head dataset is longer than encoder datast
Run "epic_cushion" (ID dev_histo_1641978736603) in innereye4ws

[AB#4839](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/4839)

### PR DESCRIPTION
**Pull request by ant0nsc**
SimCLR cosine LR scheduler was getting wrong length information when the linear head dataset was longer than the encoder dataset.
Also, removed lots of obsolete `pytest.skipif.is_windows()` annotations.

Closes #636 

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [x] Ensure that your PR is small, and implements one change.
- [x] Add unit tests for all functions that you introduced or modified.
- [x] Run PyCharm's code cleanup tools on your Python files.
- [x] Link the correct GitHub issue for tracking.
- [x] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [x] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

**Review by maxilse — APPROVED** [review]
lgtm

File: Tests/SSL/test_data_modules.py [hunk 1]
```diff
@@ -203,7 +194,8 @@ def test_combined_data_module() -> None:
                                                  data_dir=str(path_to_test_dataset),
                                                  # 300 images in total in test dataset
                                                  batch_size=60,
-                                                 shuffle=False)
+                                                 shuffle=False,
+                                                 num_workers=0)
```
Comment (vale-salvatelli) [inline Tests/SSL/test_data_modules.py:198]: Why are we changing the default num_workers? was this affecting the len issue?

Comment (ant0nsc) [reply Tests/SSL/test_data_modules.py:198]: This is not affecting the test, but the test runtime - changes runtime from 1min to 400ms :-)

File: Tests/SSL/test_ssl_containers.py [hunk 2]
```diff
@@ -124,12 +128,12 @@ def test_innereye_ssl_container_cifar10_resnet_simclr() -> None:
...
-                        'ssl_online_evaluator/val/loss': 2.268489360809326,
+    expected_metrics = {'simclr/val/loss': 2.8797268867492676,
+                        'ssl_online_evaluator/val/loss': 2.272602081298828,
                         'ssl_online_evaluator/val/AccuracyAtThreshold05': 0.20000000298023224,
-                        'simclr/train/loss': 3.6261844635009766,
+                        'simclr/train/loss': 3.6261773109436035,
                         'simclr/learning_rate': 0.0,
-                        'ssl_online_evaluator/train/loss': 3.1140503883361816,
+                        'ssl_online_evaluator/train/loss': 3.1140334606170654,
                         'ssl_online_evaluator/train/online_AccuracyAtThreshold05': 0.0}
```
Comment (vale-salvatelli) [inline Tests/SSL/test_ssl_containers.py:137]: for my own knowledge: was this due to a previous PyTorch update? nothing in this PR seems should affect these figures. If it's something that changes often maybe update the tolerance in _compare_stored_metrics?

**Review by vale-salvatelli — APPROVED** [review]
Looks good to me

File: Tests/SSL/test_ssl_containers.py (hunk 2 above)
Comment (ant0nsc) [reply Tests/SSL/test_ssl_containers.py:137]: The reason for this is a slight change to randomization. We are now creating dataloaders one more time for diagnostics.

File: InnerEye/ML/SSL/encoders.py [hunk 3]
```diff
@@ -80,9 +80,8 @@ def get_encoder_output_dim(
             SSLOnlineEvaluatorInnerEye,
         )
 
-        dataloader = dm.train_dataloader()
-        dataloader = dataloader[SSLDataModuleType.LINEAR_HEAD] if isinstance(dataloader, dict) else dataloader  # type: ignore
-        batch = iter(dataloader).next()  # type: ignore
+        batch = next(iter(dm.train_dataloader()))
+        batch = batch[SSLDataModuleType.LINEAR_HEAD] if isinstance(batch, dict) else batch  # type: ignore
```
Comment (mebristo) [inline InnerEye/ML/SSL/encoders.py:84]: Question rather than feedback: does it slow it down to index every batch rather than indexing the dataloader object as before?

Comment (mebristo) [reply InnerEye/ML/SSL/encoders.py:84]: @ant0nsc - this is the comment we were referring to

Comment (ant0nsc) [reply InnerEye/ML/SSL/encoders.py:84]: 1) This is only working on the first element of the data loader, so I would not worry too much about it
2) we can't index in the same way as before anymore because train_dataloader no longer returns a dictionary.
